lib/rubocop/cop/rails/response_parsed_body.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Prefer `response.parsed_body` to `JSON.parse(response.body)`.
+      #
+      # @safety
+      #   This cop is unsafe because Content-Type may not be
+      #   `application/json`.
```

Comment:
Is Content-Type guaranteed for code before autocorrection? I'm wondering if autocorrection will change Content-Type. So, If Content-Type doesn't change, it looks like there is no change of safety.

Reply:
It's unsafe because the behavior changes a bit.

`JSON.parse(body.response)` attempts to parse the body as JSON regardless of `Content-Type`. e.g. if the controller spits out JSON but marks it as `application/xml`, this will still work. Whereas `parsed_body` will use the appropriate parser based on `Content-Type`, and will raise an exception if the response can't be parsed.

https://github.com/rails/rails/blob/282b769fb543404187abde76ca437aa697f6d6ba/actionpack/lib/action_dispatch/testing/test_response.rb#L21-L37

Reply:
I get it. It seems to be an incompatibility behavior potential rather than a false positive potential. So it's unsafe for autocorrection.